test/it/crd-validation/etcd/validation_test.go

```diff
@@ -0,0 +1,849 @@
+			etcd.Spec.Etcd.Quota = &test.quota
+	
+			if test.backup {
+				container := "etcd-bucket"
+				provider := "Provider"
+				etcd.Spec.Backup.Store = &druidv1alpha1.StoreSpec{
+					Container: &container,
+					Provider: (*druidv1alpha1.StorageProvider)(&provider),
+					Prefix: "etcd-test",
+					SecretRef: &corev1.SecretReference{
+						Name: "test-secret",
+					},
+				}
+			}
+
+			validateEtcdCreation(t,g,etcd,test.expectErr)
+		})
+
+
+	}
+}
+
+
+// tests for the Update Validations:
+
+// etcd.spec.storageClass is immutable
+
+func TestValidateSpecStorageClass(t *testing.T) {
+	testNs, g := setupTestEnvironment(t)
+
+	
+
+	tests := []struct {
+		name string
+		etcdName string
+		initalStorageClassName string
+		updatedStorageClassName string
+		expectErr bool
+	}{
+		{
+			name: "Valid #1: Unchanged storageClass",
+			etcdName: "etcd-valid-1",
+			initalStorageClassName: "gardener.cloud-fast",
+			updatedStorageClassName: "gardener.cloud-fast",
+			expectErr: false,
+		},
+		{
+			name: "Invalid #1: Updated storageClass",
+			etcdName: "etcd-invalid-1",
+			initalStorageClassName: "gardener.cloud-fast",
+			updatedStorageClassName: "default",
+			expectErr: true,
+		},
+
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T){
+			etcd := utils.EtcdBuilderWithoutDefaults(test.etcdName, testNs).WithReplicas(3).Build()
+			etcd.Spec.StorageClass = &test.initalStorageClassName
+			
+			cl := itTestEnv.GetClient()
+			ctx := context.Background()
+			g.Expect(cl.Create(ctx, etcd)).To(Succeed())
+
+			etcd.Spec.StorageClass = &test.updatedStorageClassName
+
+
+            if test.expectErr {
+                g.Expect(cl.Update(ctx, etcd)).ToNot(Succeed())
+            } else {
+                g.Expect(cl.Update(ctx, etcd)).To(Succeed())
+            }
+		})
+	}
+}
+
+// checks the update on the etcd.spec.replicas field
+func TestValidateSpecReplicas(t *testing.T) {
+	tests := []struct {
+		name string
+		etcdName string
+		initialReplicas int
+		updatedReplicas int
+		expectErr bool
+	} {
+		{
+			name: "Valid updation of replicas #1",
+			etcdName: "etcd-valid-inc", // change later
+			initialReplicas: 3,
+			updatedReplicas: 5,
+			expectErr: false,
+		},
+		{
+			name: "Valid updation of replicas #2",
+			etcdName: "etcd-valid-zero", // change later
+			initialReplicas: 3,
+			updatedReplicas: 0,
+			expectErr: false,
+		},
+		{
+			name: "Inalid updation of replicas #1",
+			etcdName: "etcd-invalid-dec", // change later
+			initialReplicas: 5,
+			updatedReplicas: 3,
+			expectErr: true,
+		},
+	}
+
+	testNs, g := setupTestEnvironment(t)
+
+	for _, test := range tests {
+		t.Run(test.name, func (t *testing.T){
+			etcd := utils.EtcdBuilderWithoutDefaults(test.etcdName, testNs).WithReplicas(int32(test.initialReplicas)).Build()
+			cl := itTestEnv.GetClient()
+			ctx := context.Background()
+			g.Expect(cl.Create(ctx, etcd)).To(Succeed())
+
+			etcd.Spec.Replicas = int32(test.updatedReplicas)
+			time.Sleep(30)
```

Comment:
Do not use `time.sleep`,  For more detailed guidance, please refer to the
https://github.com/gardener/etcd-druid/blob/6b3b94e2b42c6d0235d6825d3ad689c3fa1c35a9/docs/development/testing.md?plain=1#L25